crates/swift-integration-tests/src/option.rs

```diff
@@ -35,10 +35,16 @@ mod ffi {
 
     extern "Rust" {
         type OptTestOpaqueRustType;
+        type OptTestOpaqueRustTypeRef;
```

Comment:
Need to rename this. the `type OptTestOpaqueRustType` above generates `class OptTestOpaqueRustTypeRef` on the Swift side, and then this also generates a class with the same name.

<img width="1025" alt="image" src="https://github.com/chinedufn/swift-bridge/assets/2099811/20dee030-95cf-40c5-a83e-27bf297454db">

Can call it `OptTestOpaqueRefRustType` instead

Reply:
ohhhhh right - that makes sense. Doh! magical strings hahaha

Reply:
Here's more information as to why this error is happening. 

No need to read this link to solve the problem.
Renaming the type (I gave an example under the screenshot in the previous comment) will do.

https://github.com/chinedufn/swift-bridge/blob/dd5bef56af28db4f1a1244d86115def6abede931/book/src/bridge-module/opaque-types/README.md?plain=1#L52-L133

Reply:
Done and pushed up

Reply:
Hmm, actually, now I'm getting this failure locally:

```
Testing failed:
	Missing argument label 'ptr:' in call
	Cannot convert value of type 'Int' to expected argument type 'UnsafeMutableRawPointer'
	'nil' requires a contextual type
	Command SwiftCompile failed with a nonzero exit code
	Testing cancelled because the build failed.
	```

Reply:
I'm taking a wild guess here as I don't know swift that well but I'm guessing it's related to my logic here:

```swift
    func testSwiftCallRustWithOptionRefOpaqueRustType() throws {
        let val = OptTestOpaqueRefRustType.new(123)
        let opt_ref = val.field_ref()

        let reflect = rust_reflect_option_ref_opaque_rust_type(opt_ref)
        XCTAssertEqual(reflect!.field(), 123)
        XCTAssertNil(rust_reflect_option_ref_opaque_rust_type(nil))
        let reflect = nil
        
        let second_reference = rust_reflect_option_ref_opaque_rust_type(opt_ref)
        XCTAssertEqual(second_reference!.field(), 123)
    }
```

I wonder if this is the problem:

```swift
        XCTAssertEqual(reflect!.field(), 123)
```

Is it saying that the result of field() is a unsafe mutable raw pointer and it's trying to coerce the int of 123 into that but failing for the equality comparison?

In this case I call `val.field_ref()` which should get us the inner Option<&Type> and then I thought `reflect!.field()` goes inside that option and calls the `field()` method on the inner type which should then get us an int out?